### ISSUE TITLE
Can't remove transported units without removing transport

### ISSUE THREAD

**Issue by drockken** [triplea-game/triplea#4446]
### Engine version

1.10.13296

### My Operating System
Windows 10/64 bit
Java 1.8.0_144-b01 according to TripleA
Java 1.8.0_121-b13 according to the Java Control Panel

### Map name
ww2global40_2nd_edition.xml

### Can you describe how to trigger the error? (eg: what sequence of actions will recreate it?)
Open Saved Game
Enter Edit Mode
Click six infantry in Dzavhan
Click Remove Selected Units, this generates the error "Can't remove transported units without removing transport"

### Do you have the exact error text? Please copy/paste if so
"Can't remove transported units without removing transport"

### Instead of this error, what should have happened?
I should have been allowed to remove the units via the edit.  There are no transports in this land-locked territory.

### Any additional information that may help
I forgot to "attack" the Russian units in Olgy so I rolled the dice after the fact and was manually going to move my units into the territory from Dzavhan and another adjacent territory.
To do this I had to enter the edit mode and manually move them.  It did not let me remove the units.

**Comment by drockken** [comment]
Once again didn't attach my Zip, trying again...
[no_transport.zip](https://github.com/triplea-game/triplea/files/2704809/no_transport.zip)

**Comment by panther2** [comment]
The first of the six Japanese Infantries can be removed as usual.
Starting from the second one this is what happens:
![tra](https://user-images.githubusercontent.com/20812775/50374472-3b328500-05ef-11e9-8e1b-06a3328d203c.PNG)

Just to illustrate...

**Comment by panther2** [comment]
I found out that the error is somehow caused by the Infantry units that came from Yenisey. As those from Tsagan Oloom can be removed without error when starting the game one round before.

Now I am tracing the history of those Yenisey units...

**Comment by ssoloff** [comment]
> Java 1.8.0_144-b01 according to TripleA
> Java 1.8.0_121-b13 according to the Java Control Panel

@drockken Just wanted to note why there is a discrepancy here...  It is likely you originally installed TripleA without having Java already installed.  In that case, the TripleA installer will download and install its own Java runtime (JRE).  Subsequent runs of TripleA (even if you update to a newer version), will _always_ use that bundled JRE no matter which version of Java you've installed via Windows Installer.

I mention this because there has been a lot of discussion about the problems these bundled JREs may cause (e.g. see #4358).  If you ever want to use the version of Java you've installed yourself, you'll need to manually delete the bundled JRE.  (Note that it's unlikely the version of Java you're using has any specific relation to this particular bug, but it's always good to be aware about these things. :smile:)

**Comment by simon33-2** [comment]
I also have noticed this problem. Glad we have a test case, or at least I think we do.

**Comment by ron-murhammer** [comment]
@panther2 @drockken I just looked through the save games and there definitely is an issue with the 'transportedBy' field not being cleared for the infantry that amphibious assault Korea in round 6 (nice work tracing that back @panther2). I was able to reproduce it as well by taking a save before that battle and making the same attack on korea: 
[test_after_cm.zip](https://github.com/triplea-game/triplea/files/2717523/test_after_cm.zip). After rolling that battle if I try to edit out those infantry left in Korea then I get the error.

I should be able to look through the code to see why that is happening.

**Comment by simon33-2** [comment]
Given that battles with no defenders are now fought automatically in BattleDelegate.start(), is there a need for the concept of a finished battle?

**Comment by ron-murhammer** [comment]
@simon33-2 It appears to be used to mark territories that were blitzed/conquered for this XML property:
```
battle	values: checks if there has been a battle in the territory. Format is "attacker:defender:battleType:rounds:territory1", where attacker, defender, and battleType can all be "any", and rounds can be "currentRound". Territories can be any number of territories, and if more than one territory is listed then if any of them are true, it will return true.  You can also have multiple instances
example: <option name="battle" value="any:Japanese:any:2-3:Manchuria"/> or <option name="battle" value="Russians:Japanese:any:currentRound:Manchuria:Jehol:Chahar:Suiyuyan:Kansu"/>
```

**Comment by veqryn** [comment]
I believe FinishedBattle is also around to provide a history of conquered and blitzed territories, in addition to being used by various triggers and attachments.

I'm leaning towards option number 2 above: have FinishedBattle clear the transportedBy.  Option 1 might be better or worse, but I'm unsure of what other ramifications it might have, so unless you all have added a lot more test cases for all the games' rule's logics', then I'm unsure about it.  Option 3 and 4 would definitely affect a lot of games that use convoys, blockades, etc etc, so I'd probably not do those.

But I've also been out of the code for years at this point, so do whatever makes sense.

**Comment by ron-murhammer** [comment]
@veqryn Thanks for the input. Just wanted to see if you remembered anything around FinishedBattles since you had implemented them. I'll probably move forward with option 1 or 2 after I look over it a bit more.

**Comment by ron-murhammer** [comment]
@drockken Once #4475 is merged, this issue should be fixed in the latest pre-release. It won't fix your existing game but should prevent "Can't remove transported units without removing transport" from occurring for non-transported units in the future.
